tests/typecheck/managers/querysets/test_as_manager.yml

```diff
@@ -64,13 +66,16 @@
         -   path: myapp/__init__.py
         -   path: myapp/models.py
             content: |
+                from typing import TypeVar
                 from django.db import models
 
-                class MyQuerySet(models.QuerySet):
+                M = TypeVar("M", bound=models.Model, covariant=True)
+
+                class MyQuerySet(models.QuerySet[M]):
                     ...
 
                 class MyModel(models.Model):
-                    objects = MyQuerySet.as_manager()
+                    objects = MyQuerySet.as_manager()  # type: ignore[var-annotated]
```

Comment:
I am not quite sure how to fix this.  the error here is sorta valid but "fixed" later in the cycle

I think it's because `as_manager` is implemented using the wrong mypy plugin hook (it's using the dynamic class hook -- but this call actually returns an _instance_ -- I think it should be the function hook instead, but I'm not sure if that can add classes that late?)

Reply:
I think dynamic class hook is the right time for creating the manager type from `as_manager`. But I find it off that the hook overwrites/inserts a new `Var`. It should be enough to point the type of an existing `Var` to the manager type the hook creates

Reply:
I'm not so sure -- the example in the plugin docs is about building a class `X`: https://github.com/python/mypy/blob/e67decb91ea5aacbb1e126463b02d78ded680a90/mypy/plugin.py#L800

but I'll see what I can do

Reply:
moved the `as_manager` instance construction to a function hook (but left the classdef hook mostly alone beyond removing the problematic `Var` insertions) -- fixed the problem \o/